authors/apps/articles/views.py

```diff
@@ -111,3 +113,99 @@ def delete(request, slug, **kwargs):
             return Response({"message": result}, status.HTTP_400_BAD_REQUEST)
 
         return Response({"message": result}, status.HTTP_200_OK)
+
+
+class ShareArticleView(generics.RetrieveAPIView):
+    """
+    This class holds an endpoint that allows authenticated users to share
+    articles across other social media platforms
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Get article endpoint with social media links to share their articles
+        """
+
+        # Gets the provider in the url
+        provider = kwargs['provider']
+        context = {'request': request}
+
+        try:
+            article = Articles.objects.get(slug=kwargs['slug'])
+
+        except Exception as error:
+            print('Received error is : {}'.format(error))
+            return Response({"message": "Article does not exist"},
+                            status.HTTP_404_NOT_FOUND)
+
+        # building links for the user on social media website
+        article_url = request.build_absolute_uri(
+            reverse('articleSpecific', kwargs={'slug': article.slug})
+        )
+
+        share_link = self.get_link(context, provider, article, article_url)
+
+        if not share_link:
+
+            return Response(
+                {
+                    'error': 'provider link is invalid',
+                }, status.HTTP_400_BAD_REQUEST)
+
+        return Response(
+            {
+                'share': {
+                    'link': share_link,
+                    'provider': provider
+                }
+            }, status.HTTP_200_OK)
+
+    def get_link(self, context, provider, article, article_url):
```

Comment:
There are quite a number of if statements with their underlying logic. Consider having a separate class with the logic for each of the providers in a separate function eg have get_facebook_url, get_twitter_url etc.

then create a dictionary with the providers as the key and the function as the value i.e 
```
providers = {
"twiter":get_twiter_url,
"facebook" : get_facebook_url,
"linked_in" : get_linked_in_url,
}
```
in your view have function  that takes the provider and applies the appropriate function in the dictionary e.g

```
def share_url(provider,**kwargs):
    for key, value in providers:
        if key == provider:
               value(**kwargs)

```

This will reduce the if statements and make your code more OOP oriented